src/lib/UI/Value/ValueFactory.php

```diff
@@ -180,17 +195,31 @@ public function createRelation(Relation $relation, Content $content): UIValue\Co
      *
      * @throws \eZ\Publish\API\Repository\Exceptions\NotFoundException
      * @throws \eZ\Publish\API\Repository\Exceptions\UnauthorizedException
+     * @throws \eZ\Publish\API\Repository\Exceptions\BadStateException
      */
     public function createRelationItem(RelationListItem $relationListItem, Content $content): UIValue\Content\Relation
     {
         $contentType = $content->getContentType();
         $relation = $relationListItem->getRelation();
         $fieldDefinition = $contentType->getFieldDefinition($relation->sourceFieldDefinitionIdentifier);
 
+        $contentInfo = $content->contentInfo;
+
+        try {
+            $relationLocation = $this->locationService->loadLocation($contentInfo->mainLocationId);
+        } catch (UnauthorizedException $e) {
+            // try different locations if main location is not accessible for the user
+            $relationLocations = $this->locationService->loadLocations($contentInfo);
+            if (empty($relationLocations)) {
+                throw $e;
+            }
+            $relationLocation = reset($relationLocations);
+        }
```

Comment:
duplicated code fragment

Reply:
+1, we should be able to reuse this logic in other places to stay consistent and be able to make changes to the fallback strategy easily.